### PR TITLE
f_william_pwa_workbox_webpack_plugin

### PR DESCRIPTION
**Pull request by RiftSplitter**
- Treats now able to support workbox webpack plugin 3.6.3
- Added service worker register in client
- Added plugin option for treats.config.js

### CODE REVIEW

File: packages/treats/scripts/config/util/workbox-generator.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const {
+        pluginMode,
+        options
+    } = workboxConfig;
+
+    if (!(pluginMode in WorkboxWebpackPlugin)) {
+        console.error(`Workbox error: ${pluginMode} is not a valid workbox plugin mode`);
+        return workboxPlugin;
+    }
+    else {
```
Comment (regalius) [inline packages/treats/scripts/config/util/workbox-generator.js]: Hi, did you mean else`if(pluginMode === "InjectManifest") {` ?

**Comment by regalius** [comment]
Hi @RiftSplitter,

Great job on implementing the `WorkboxWebpackPlugin` 👍,

I think it would be a nice improvement if user can set their own path for `service-worker.js` file via `swDest` option, another nice improvemnet is if we can add an attribute to `service-worker.js` file's script tag so we can just get the `src` from that tag, this way we won't have to add another parameter to `initClient` for the service worker file path


Thanks and great work!

File: packages/treats/scripts/config/util/workbox-generator.js (hunk 1 above)
Comment (RiftSplitter) [reply packages/treats/scripts/config/util/workbox-generator.js]: I guess I can make it better like this perhaps:
`if (!(pluginMode in WorkboxWebpackPlugin)) {
        console.error("");
        return workboxPlugin;
 } else if (pluginMode === "InjectManifest") {
        const { swSrc } = options;
        
        if (!swSrc || swSrc === "")  { 
            console.error("");
            return workboxPlugin;
}
workboxPlugin = [new WorkboxWebpackPlugin[pluginMode](options)];
return workboxPlugin;
'

**Comment by martinock** [comment]
Nice indeed @RiftSplitter ! It's also good to create the **docs** and **example app** for developers, since they can customise it on their own.
You may simply referring to workbox docs page if the config are the same. Don't forget to mention that we can add the config in `treats.config.js`.

**Pull request closed without merging**